src/main/java/org/citeplag/MathController.java

```diff
@@ -48,26 +55,60 @@
 @RestController
 @RequestMapping("/math")
 public class MathController {
-
+    // Header entry to provide hash values in the response
+    private static final String HEADER_HASH_KEY = "x-resource-location";
+    // Logging via Log4j2
     private static Logger logger = Logger.getLogger(MathController.class);
-
-    @Autowired
-    private LateXMLConfig lateXMLConfig;
+    // Caches for Mathoid Requests
+    private static HashMap<String, String> requestHashesTable = new HashMap<>();
+    private static HashMap<String, MMLEndpointCache> responseHashesTable = new HashMap<>();
```

Comment:
If you use a simple hash map as a cache better wrap it with Collections.synchronizedMap() or ConcurrentHashMap.
These tables are never cleaned, right?